rackcity/utils/rackcity_utils.py

```diff
@@ -275,24 +275,54 @@ def empty_vs_null_comparison(existing_value, new_value):
 
 
 def no_infile_location_conflicts(asset_datas):
-    location_occupied_by = {}
+    rack_location_occupied_by = {}
+    chassis_slot_occupied_by = {}
     unnamed_asset_count = 0
     for asset_data in asset_datas:
-        rack = asset_data["rack"]
-        height = ITModel.objects.get(id=asset_data["model"]).height
-        rack_position = int(asset_data["rack_position"])
-        asset_location_range = [  # THIS IS REPEATED! FACTOR OUT.
-            rack_position + i for i in range(height)
-        ]
-        if rack not in location_occupied_by:
-            location_occupied_by[rack] = {}
-        for location in asset_location_range:
-            if location in location_occupied_by[rack]:
+        if asset_data["offline_storage_site"]:
+            continue
+        model = ITModel.objects.get(id=asset_data["model"])
+        if model.is_rackmount():
+            rack = asset_data["rack"]
+            height = model.height
+            rack_position = int(asset_data["rack_position"])
```

Comment:
are these values guaranteed to exist at this point in the codepath?

Reply:
Iff the file is error free. Error handling in bulk is not amazing